extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -290,7 +290,7 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, logger l
 			},
 		}
 
-		if _, err := controllerutil.CreateOrUpdate(ctx, a.client, machineDeployment, func() error {
```

Comment:
> Would that mean that all currently existing other Taints will be overwritten after this change?

Yes, it will overwrite all lists contained here as a whole. But that's fine, because the spec is supposed to be exclusively owned by the worker controller and no other controller is supposed to make changes to it. Even if some other controller does modify the spec, eventually overwriting it again is the right thing to do, as the worker controller knows the exact desired state, which is independent of other (concurrent) modifications, and the object should eventually converge to it again.

Reply:
Good to know. Then my understanding was correct that we have a change in behaviour but we accept that here.